src/board.js

```diff
@@ -0,0 +1,73 @@
+import { useState } from 'react'
+import { Square } from './square.js'
+
+const board = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
+
+export function Board(props) {
+  const [squareColors, setSquareColors] = useState(Array(board.length * board.length).fill('grey'))
+  const [clickedSquare, setClickedSquare] = useState(null)
+
+  const handleSquareClick = (i) => {
+    const squares_slice = squareColors.slice()
+    squares_slice[i] = 'red'
+    setSquareColors(squares_slice)
+    setClickedSquare(i)
+  }
+
+  const handleSquareHover = (i) => {
+    if (clickedSquare) {
+      const [hoveredRow, hoveredColumn] = squareIndexToRowColumn(i)
+      const [clickedRow, clickedColumn] = squareIndexToRowColumn(clickedSquare)
+
+      if (hoveredRow === clickedRow) {
+        const squares_slice = squareColors.slice()
+        squares_slice[i] = 'red'
+        setSquareColors(squares_slice)
+      }
+
+      if (hoveredColumn === clickedColumn) {
+        const squares_slice = squareColors.slice()
+        squares_slice[i] = 'red'
+        setSquareColors(squares_slice)
+      }
+    } 
+  }
+
+  const handleMouseUp = (_e) => setClickedSquare(null)
+
+  const squareRowColumnToIndex = (row, column) => row * board.length + column
+
+  const squareIndexToRowColumn = (i) => {
+    const row = Math.floor(i / board.length)
+    const column = i % board.length
+    return [row, column]
+  }
+
+  const renderSquare = (row, column) => {
+    const i = squareRowColumnToIndex(row, column)
+
+    return (
+      <Square
+        key = {i}
+        color={squareColors[i]}
+        onClick={() => handleSquareClick(i)}
```

Comment:
If you don't need to pass props, you dont need to add ().
so it can just be: onClick={handleSquareClick(i)}

Reply:
Is that right? Won't `onClick` expect a function with one arg?

Reply:
nope the event is just ignored. if you dont need to capture the contents of the event itself, then theres no need to wrap the function you're trying to call in an extra function that calls it